css/style.css

```
@@ -0,0 +1,950 @@
+  width: 50%;
+}
+
+.column-3 {
+  width: 100%;
+}
+
+.column-4 {
+  width: 100%;
+}
+
+.column-6 {
+  width: 100%;
+}
+
+.column-8 {
+  width: 100%;
+}
+
+@media only screen and (min-width: 768px) {
+  /* For tablet: */
+  .column {
+    float: left;
+  }
+
+  .column-2 {
+    width: 25%;
+  }
+
+  .column-3 {
+    width: 33.3%;
+  }
+
+  .column-4 {
+    width: 50%;
+  }
+
+  .column-6 {
+    width: 50%;
+  }
+
+  .column-8 {
+    width: 50%;
+  }
+}
+
+@media only screen and (min-width: 1024px) {
+  /* For desktop: */
+  .column {
+    float: left;
+  }
+
+  .column-2 {
+    width: 20%;
+  }
+
+  .column-3 {
+    width: 25%;
+  }
+
+  .column-4 {
+    width: 33.3%;
+  }
+
+  .column-6 {
+    width: 50%;
+  }
+
+  .column-8 {
+    width: 66.6%;
+  }
+}
+
+/*
+ * FLEX
+ */
+.flex {
+  display: flex;
+}
+
+.flex-1 {
+  flex: 1;
+}
+
+.v-align-top {
+  align-items: flex-start;
+}
+
+.v-align-center {
+  align-items: center;
+}
+
+.v-align-bottom {
+  align-items: flex-end;
+}
+
+.h-align-left {
+  justify-content: flex-start;
+}
+
+.h-align-center {
+  justify-content: center;
+}
+
+.h-align-right {
+  justify-content: flex-end;
+}
+
+/*
+ * MARGINS
+ */
+.margin-auto {
+  margin: auto;
+}
+
+.margin-sm {
+  margin: 0.6rem;
+}
+
+.margin-md {
+  margin: 0.8rem;
+}
+
+.margin-lg {
+  margin: 1rem;
+}
+
+.margin-top-auto {
+  margin-top: auto;
+}
+
+.margin-top-sm {
+  margin-top: 0.6rem;
+}
+
+.margin-top-md {
+  margin-top: 0.8rem;
+}
+
+.margin-top-lg {
+  margin-top: 1rem;
+}
+
+.margin-bottom-auto {
+  margin-bottom: auto;
+}
+
+.margin-bottom-sm {
+  margin-bottom: 0.6rem;
+}
+
+.margin-bottom-md {
+  margin-bottom: 0.8rem;
+}
+
+.margin-bottom-lg {
+  margin-bottom: 1rem;
+}
+
+.margin-left-auto {
+  margin-left: auto;
+}
+
+.margin-left-sm {
+  margin-left: 0.6rem;
+}
+
+.margin-left-md {
+  margin-left: 0.8rem;
+}
+
+.margin-left-lg {
+  margin-left: 1rem;
+}
+
+.margin-right-auto {
+  margin-right: auto;
+}
+
+.margin-right-sm {
+  margin-right: 0.6rem;
+}
+
+.margin-right-md {
+  margin-right: 0.8rem;
+}
+
+.margin-right-lg {
+  margin-right: 1rem;
+}
+
+@media only screen and (min-width: 768px) {
+  /* For tablet: */
+  .margin-sm {
+    margin: 0.6rem;
+  }
+
+  .margin-md {
+    margin: 0.8rem;
+  }
+
+  .margin-lg {
+    margin: 1rem;
+  }
+
+  .margin-top-sm {
+    margin-top: 0.6rem;
+  }
+
+  .margin-top-md {
+    margin-top: 0.8rem;
+  }
+
+  .margin-top-lg {
+    margin-top: 1rem;
+  }
+
+  .margin-bottom-sm {
+    margin-bottom: 0.6rem;
+  }
+
+  .margin-bottom-md {
+    margin-bottom: 0.8rem;
+  }
+
+  .margin-bottom-lg {
+    margin-bottom: 1rem;
+  }
+
+  .margin-left-sm {
+    margin-left: 0.6rem;
+  }
+
+  .margin-left-md {
+    margin-left: 0.8rem;
+  }
+
+  .margin-left-lg {
+    margin-left: 1rem;
+  }
+
+  .margin-right-sm {
+    margin-right: 0.6rem;
+  }
+
+  .margin-right-md {
+    margin-right: 0.8rem;
+  }
+
+  .margin-right-lg {
+    margin-right: 1rem;
+  }
+}
+
+@media only screen and (min-width: 1024px) {
```

margin-sm means margin small screens only. margin-md means margin for medium screens only. You need to seperate these properties along with others(e.g border, padding, etc.) according to the screen width.
```suggestion
@media only screen and (max-width: 720px) {
 /* should only contain properties that works for small screen*/
 margin-sm-Nth {/*span elements*/}
 border-sm-Nth {/*span elements*/}
}
@media only screen and (min-width: 721px,max-width: 1024px) {
 /* should only contain properties that works for medium screens*/
 margin-md-Nth {/*span elements*/}
 border-md-Nth {/*span elements*/}
}
```
You get the point Right?

---

I see your point, however, we do have different margins and borders for different screen sizes. We used media queries instead of class names, so perhaps that's causing confusion. Is this method acceptable?

---

You do have them for different screen sizes. However, you cannot append "md" to a property that goes inside a media query for small screen
```suggestion
@media only screen and (max-width: 720px) {
  /* should only contain properties that works for small screen*/
  .margin-sm-Nth {/*span elements*/}/* cool*/
  .border-sm-Nth {/*span elements*/}/* cool*/
  .border-md-Nth {/*span elements*/}/* **NOT Cool***/
   }
   @media only screen and (min-width: 721px,max-width: 1024px) {
   /* should only contain properties that works for medium screens*/
  .margin-md-Nth {/*span elements*/}/* cool*/
  .border-md-Nth {/*span elements*/}/* cool*/
   }
```